src/ExceptionTransformer.ts

```diff
@@ -46,9 +68,127 @@ class ExceptionTransformer {
+        }
+      }
+      and return `["a logical message"]`
+
+      if value of the `fieldName` is not string[], 
+      it will generate the first meaningful message and return it as string[]
+  */
+  generateSpecificFieldError(
+    errorInfo: Exception | null | undefined
+  ): (fieldName: string) => string[] | undefined {
+    const errorDetail = getErrorDetail(errorInfo);
+
+    if (errorDetail) {
+      return function getError(fieldName: string) {
+        let fieldError: string[] | undefined;
+        //fieldName can be string only
+        if (typeof fieldName === "string") {
+          const errorValue = getValueFromPath(errorDetail, fieldName);
+
+          // errorValue can be string[], ExceptionDetail[], ExceptionDetail or undefined
+          if (errorValue) {
+            if (isArrayOfString(errorValue)) {
+              fieldError = errorValue as string[];
+            } else {
+              fieldError = getStringMessage(errorValue)
+                ? [getStringMessage(errorValue)]
+                : undefined;
+            }
+          }
+        }
+        return fieldError;
+      };
+    }
+
+    if (errorInfo && this.onUnexpectedException) {
+      this.onUnexpectedException({
+        type: "NO_ERROR_DETAIL",
+        error: undefined,
+        errorInfo
+      });
+    }
+
+    return () => undefined;
+  }
+
+  /*  Generates a printable error message.
+   *   `options`:
+   *            `skipTypes`: (array of strings) Error types to be skipped.
+   *             `knownErrorKeys`: (array of strings) Error keys to be skipped inside `Exception["detail"]`.
+   *
+   *   Returns "";  when all error keys in errorInfo.detail are known (that is they are included in `knownErrorKeys`) or errorInfo.type should skipped.
+   *   Returns first meaningful string found in non_field_errors, if it exists.
+   *   Returns `${key}: ${value}`; When there is no non_field_errors, returns the first unknown key and its value combined in a string.
+   *   Returns fallback_message; when it couldn't generate any meaningful message using the methods above.
+   *   Returns genericErrorMessage; when there is no fallback_message
+   */
+  generateErrorMessage(errorInfo: Exception, options: Options = {}): string {
+    const { knownErrorKeys = [], skipTypes = [] } = options;
+    const shouldSkipError = skipTypes && skipTypes.includes(errorInfo.type);
+    let finalMessage = "";
+
+    try {
+      if (!shouldSkipError) {
+        let errorDetail = getErrorDetail(errorInfo);
+        let shouldDisplayFallbackMessage = false;
+        let message = "";
+
+        if (errorDetail && !isObjectEmpty(errorDetail)) {
+          if (knownErrorKeys && knownErrorKeys.length) {
+            errorDetail = knownErrorKeys.reduce((object, errorKey) => {
+              // delete all `knownErrorKeys` from errorDetail
+              return deleteProperty(object, errorKey);
+            }, errorDetail);
+          }
+          message = getStringMessage(errorDetail);
```

Comment:
```suggestion
          message = getStringMessage(removeKnownKeysFromErrorDetail(errorDetail));
```

The if block above can be separated into a utility and used like this